### PR TITLE
Use service monitor

### PR DESCRIPTION
**Pull request by QuentinBisson**
This configures the operator to be scraped using a service monitor instead of the old scheme

### CODE REVIEW

**Review by weseven — APPROVED** [review]

**Comment by Gacko** [comment]
Your PR says it makes use of a PodMonitor while you're actually using a ServiceMonitor? IMHO we can drop the service and really just use a PodMonitor as you proposed for the `etcd-kubernetes-resources-count-exporter`.

**Review by Gacko — CHANGES_REQUESTED** [review]
Your PR says it makes use of a PodMonitor while you're actually using a ServiceMonitor? IMHO we can drop the service and really just use a PodMonitor as you proposed for the `etcd-kubernetes-resources-count-exporter`.

**Comment by QuentinBisson** [comment]
I tried but it does not work because this pod is using hostnetwork:true

**Comment by Gacko** [comment]
But that's probably more an issue because of not using CiliumNetworkPolicies, right? Like, plain NetworkPolicies can not be used for allowing traffic to host network.

**Comment by QuentinBisson** [comment]
Hmm thé issue I had was really related to kyverno policy exceptions but you're free to try a Pod monitor instead :) I don't have the bandwith to do them all "properly", I really just need to get rid of the old annotations asap:)

File: helm/etcd-backup-operator/templates/service.yaml [hunk 1]
```diff
@@ -5,12 +5,9 @@ metadata:
...
-  annotations:
-    prometheus.io/scrape: "true"
 spec:
   ports:
   - name: metrics
-    port: {{ .Values.service.port }}
+    port: {{ .Values.port }}
   selector:
-    {{- include "labels.selector" . | nindent 4 }}
+    {{- include "labels.selector" . | nindent 4 }}
```
Comment (Gacko) [inline helm/etcd-backup-operator/templates/service.yaml:13]:
```suggestion
    {{- include "labels.selector" . | nindent 4 }}
```

Adding the missing/removed newline at EOF.

**Review by Gacko — APPROVED** [review]

**Review by Gacko — CHANGES_REQUESTED** [review]
Ah, one more: Then please do not change the values (replace `service.port` by just `port`) if you keep using a ServiceMonitor.

**Comment by QuentinBisson** [comment]
Sure :)

**Comment by QuentinBisson** [comment]
It's fixed @Gacko

**Review by Gacko — APPROVED** [review]

**Pull request merged**